scripts/src/saforcharttesting/saforcharttesting.py

```diff
@@ -156,6 +156,23 @@ def write_sa_token(namespace, token):
                 with open(token, "w") as fd:
                     fd.write(base64.b64decode(content).decode("utf-8"))
 
+def switch_project_context(namespace, token):
+    tkn = open(token).read()
+    for i in range(7):
```

Comment:
why do we set limit to 7 iterations? How do we know this is enough and what happens if it is not?

Reply:
I noticed the resource becomes available within 1 minute. Currently, the maximum waiting time is 7x10 = 70 seconds.
I think we can exit the program with error code 1. so that the job is going to fail.  This should be considered an infra failure. I will write a detailed error message to help the person who reads it.

Reply:
Added program exit with error message: 89ef26372c26613c564f3613ba19fa78eb85fa49